test/test_program.cpp

```diff
@@ -0,0 +1,56 @@
+//Here, the programs of kernels are tested.
+#include "yacx/Headers.hpp"
+#include "yacx/KernelArgs.hpp"
+#include "yacx/Source.hpp"
+#include "yacx/Program.hpp"
+
+#include <catch2/catch.hpp>
+#include <iostream>
+
+using yacx::Program, yacx::KernelArg, yacx::Source, yacx::Header, yacx::Headers;
+
+TEST_CASE("A program is created, compiled, configured and then launched."){
```

Comment:
if we look at the [header file](https://github.com/ZerataX/yacx/blob/master/include/yacx/Program.hpp) for Program, we see that Program only really has 2 functions: 
`compile` and `instantiate`, so that is the only thing we really need to test.

  - for `compile` it's really only interesting to test compiling source code with wrong syntax, source code that requires compiler options, e.g. `--std=c++14` and syntactically correct source code.
  - for `instantiate` a few common template use cases and some where the function is used wrong would be great.

Not sure if most compiler options should be tested here or in separate test for compiler options